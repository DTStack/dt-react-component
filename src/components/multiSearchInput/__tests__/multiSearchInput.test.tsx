import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import MulSelectDropdown from '../index';

describe('mulSelectDropdown Component test', () => {
    afterEach(() => {
        cleanup();
    });
    it('should render mulSelectDropdown correctly', () => {
        const { container: wrapper } = render(
            <MulSelectDropdown searchType="tail" value="this is value" placeholder="hello" />
        );
        expect(wrapper.firstChild).toMatchSnapshot();
    });
    it('custom filter rendering', () => {
        const { queryByTestId } = render(<MulSelectDropdown filterOptions={['front', 'tail']} />);
        expect(queryByTestId('icon-front')).not.toBeNull();
        expect(queryByTestId('icon-tail')).not.toBeNull();
        expect(queryByTestId('icon-caseSensitive')).toBeNull();
        expect(queryByTestId('icon-precise')).toBeNull();
    });
    it('input value change', () => {
        const myMockChange = jest.fn((value) => value);
        const { queryByTestId } = render(
            <MulSelectDropdown searchType="tail" onChange={myMockChange} />
        );
        fireEvent.change(queryByTestId('input'), { target: { value: '1234567891011' } });
        expect(myMockChange).toHaveBeenCalled();
        expect(myMockChange.mock.results[0].value).toBe('1234567891011');
    });
    it('input search', () => {
        const myMockSearch = jest.fn((value: any, searchType: any) => {
            return { value, searchType };
        });
        const { queryByTestId } = render(
            <MulSelectDropdown
                searchType="tail"
                value="this is value"
                placeholder="hello"
                onSearch={myMockSearch}
            />
        );
        fireEvent.keyDown(queryByTestId('input'), { key: 'Enter', keyCode: 13 });
        expect(myMockSearch.mock.calls.length).toBe(1);
    });
    it('should  support MulSelectDropdown custom className', () => {
        render(
            <MulSelectDropdown
                searchType="tail"
                value="this is value"
                placeholder="hello"
                className="testMulSelectDropdown"
            />
        );
        const ele = document.querySelector('.testMulSelectDropdown');
        expect(ele).not.toBeNull();
    });
});
