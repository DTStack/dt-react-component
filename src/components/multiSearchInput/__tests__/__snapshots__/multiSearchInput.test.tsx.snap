// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mulSelectDropdown Component test should render mulSelectDropdown correctly 1`] = `
<div
  class=""
  style="position: relative; width: 250px;"
>
  <input
    class="ant-input"
    data-testid="input"
    placeholder="hello"
    style="padding-right: 83px;"
    type="text"
    value="this is value"
  />
  <div
    style="position: absolute; height: 100%; top: 0px; right: 0px; width: 83px; display: flex; justify-content: space-around; align-items: center; padding-right: 5px;"
  >
    <div
      data-testid="icon-precise"
      style="cursor: pointer; display: block; height: 22px; width: 26px; overflow: hidden;"
    >
      <img
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAMAAABEpIrGAAAAwFBMVEUAAAD///+AgICioqKSkpKcnJyUlJSbm5uVlZWXl5eWlpaWlpaYmJiZmZmWlpaVlZWVlZWWlpaVlZWYmJiWlpaYmJiWlpaXl5eYmJiXl5eXl5eWlpaYmJiYmJiXl5eYmJiYmJiWlpaXl5eXl5eWlpaWlpaXl5eWlpaWlpaYmJiXl5eXl5eYmJiYmJiXl5eYmJiXl5eWlpaXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eIXQHmAAAAP3RSTlMAAQILDhITFxggIi4vMjM1Oj9BQ0RISVFUXWBmam9zdHmBhIWGjZaXoaOkp6qvvL7EyMzR2t3g4eTn6erz+PmloSGXAAAAAWJLR0QB/wIt3gAAAJlJREFUOMtjYBgF1AUqdvYowE4ZTYEdK4QW1oPQbLZoCuyhNJ82mgCKAh1TRgwdyHw2M31BIMUtxo5DgayqpAYDg5yBsjknFgX2JgzGPMxWLAxcAuL6kpb2WEzgtzEyspZmUNOVMZDCaoW6PAODiCGDqSi/BXYFChwMDExKzEJaihJ8OLwJBLyaOLxpx0YgJJXR40JpNH2SBgAKqha54oD4rgAAAABJRU5ErkJggg=="
        style="margin-top: -6px; margin-left: -3px;"
      />
    </div>
    <div
      data-testid="icon-front"
      style="cursor: pointer; display: block; height: 22px; width: 26px; overflow: hidden;"
    >
      <img
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABmJLR0QA/wD/AP+gvaeTAAAB20lEQVRYhe2QPYgTURSFz33zJtkogrI6hcUqIm5pobiVVlsq6SwEmyQwZMCfThYRXiXYSZjMzdOwYVER0gsWay2ynYiFICgoGIw2oilm8q7NBpYVMkEN28xX3nM553CAgoKCgoIpMPPLOI4X55mhcvSjvu/v38sCc2emAsYY1ev1Fnbfmfk1M2/NtYCIlIMgeDYaja7tCj8HoCQii51O5/ROrd/ve0mSnLDWHv4fBZ4qpcae5/WSJDlkjNHbUo2INgA8EpHa5D+O42PD4fAtgA3n3CtmNv9UAMAZEbngnHtHRO+DILhnrd0H4DIRPVFKPQZwpdVqlQFAa30EwK0ois5rrVcB3JxmrqeJACAiz5VSZSK6GIbhLwBg5qsAlIjc3n5b8H2/CqCfZdlXz/PuMvNalmUZgAPT/HMXIKImgA8isrbjXCeiB865TefcJoB1IqoDgFLqDoCtZrO5QkQ38vxzF1BKuTAMa8YYBQDtdvukiJwtlUqX6vX6DwDodrsv0jT9ZK1dGo/Hb0TkOjMfFJHjANxU/7wCE4wxE6MvRLQ8CQeARqPx3Tm3XKlUvkVRdN/zvKpz7uFgMKhqrU/NmvEHzPzRWrv01wYzkLfA5zRNf86zQEFBQcGe8xvNVbTHyF7uqwAAAABJRU5ErkJggg=="
        style="margin-top: -6px; margin-left: -3px;"
      />
    </div>
    <div
      data-testid="icon-tail"
      style="cursor: pointer; display: block; height: 22px; width: 26px; border: 1px solid #2491f7; overflow: hidden;"
    >
      <img
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABmJLR0QA/wD/AP+gvaeTAAAB9klEQVRYhe2UP4gTQRjF37fJJqcRDw0kEUTwDygIIgh2NoIoWFwjKU5QDIRjJhxoddptY32wxcyuWEi4QnK1eJ1a2HqNIAh3V6jkApcyISCzz0YhitmcSqz2B9PM+2bem2+GATIyMjIyZkwURctRFC1P0vOzDkCynKZ7sw4wjakBrLUPrLU0xlz4k43b7XZpP3X7uYIGyQ0RaQC4/0u4inPuQKVS+Vyv192P+TiOjw0Gg01jzM1pm6d2wFp7GcCc7/sawGIYhsUx7QmAt57nrfX7/c0wDA8DgDHmiHNuRHJFRDYAXPzrAAAaANrNZnMHwAff9xcAoNPpFEhulcvl81rrKyKy5/v+NWPMoohsfR+rAI4CWEgzkElCHMcHkyTpAngtIrskLwHYU0rdICnW2scicp3kVxE5CWBFKfVsrENLAB6RfCMiO0qp4Hc+E98AyVsAPpJcIwkReUXyaRzHJ4wxxz3Pu+153plut+uq1eq78bXGmFMAHpK8KiJ30jow8QpINkjGWut1rfW6Uuq5iLxIkuQugE8ADjnnbK1WeykiPx1Ea73d6/VOa62308xTO5DP5+/Nz89/GZ8bjUZLhUJhrtVq7YZheK5YLJ4dDofvc7mcXyqVRuO1QRAk08z/C9bawFobTNJn/hWLSH/WHhkZGRn/xDfA/cJ9mTtA3QAAAABJRU5ErkJggg=="
        style="margin-top: -6px; margin-left: -3px;"
      />
    </div>
  </div>
</div>
`;
